rust/src/api/get_node_by_path.rs

```diff
@@ -0,0 +1,118 @@
+use crate::schema::model::Schema;
+use std::fs;
+
+static DATA_FOLDER: &'static str = "src/_data";
+static CARGO_MANIFEST_DIR: &'static str = env!("CARGO_MANIFEST_DIR");
+
+pub fn get_node_by_path(path: &str) -> Option<Schema> {
+    let fs_path = format!("{}/{}/{}/info.json", CARGO_MANIFEST_DIR, DATA_FOLDER, path);
+    let Ok(info) = fs::read_to_string(fs_path) else {
+        return None;
+    };
+    let Ok(schema) = serde_json::from_str::<Schema>(info.as_str()) else {
+        return None;
+    };
+    Some(schema)
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
Can you put the content of this mod to a `test.rs` without the surrounding `mod test`?

Reply:
it seems the convention is to write unit tests within the code file, see [here](https://doc.rust-lang.org/book/ch11-03-test-organization.html#unit-tests). will skip for now since this file is gonna change drastically in the next PRs, will see how it looks then and decide